### PR TITLE
Add network composition to Squadron-compose

### PR DESCRIPTION
**Pull request by lpzimmermann**
When building a container you can now define networks that the containers should be in.
This allows connections between containers.
**There are a lot of formatting changes, sorry for that. They are made with the Swiss Life approved formatter :)**

- [x] Extend ContainerSettings
- [x] Implement networking functionality
- [x] Add ability for Squadron.Mongo
- [x] Write tests

### CODE REVIEW

File: src/Compose/ComposeResource.cs [hunk 1]
```diff
@@ -1,14 +1,14 @@
-using System;
-using System.Collections.Generic;
+using System;
```
Comment (lpzimmermann) [inline src/Compose/ComposeResource.cs:1]: Only formatting changes, can be reverted

File: src/Compose/ComposeResourceBuilder.cs [hunk 2]
```diff
@@ -29,16 +29,18 @@ private ComposableResourceBuilder(string name, ComposableResourceType resourceTy
         }
 
         public ComposableResourceBuilder<TResourceOptions> AddLink(
-            string name,
+            string linkContainerName,
```
Comment (lpzimmermann) [inline src/Compose/ComposeResourceBuilder.cs:32]: Can be reverted

File: src/Core/ContainerResourceBuilder.cs [hunk 3]
```diff
@@ -96,79 +96,106 @@ public ContainerResourceBuilder AddCmd(params string[] cmd)
...
+            _options.Networks.Add(network);
+            return this;
+        }
+
+        /// <summary>
+        /// Adds networks of which the container should be part of.
+        /// </summary>
+        /// <param name="networks">The network names.</param>
+        /// <returns></returns>
+        public ContainerResourceBuilder AddNetworks(
```
Comment (lpzimmermann) [inline src/Core/ContainerResourceBuilder.cs:185]: Remove this method and iterate from outside this method over the network list

File: src/Core/UniqueNameGenerator.cs [hunk 4]
```diff
@@ -1,14 +1,21 @@
...
-using System.Collections.Generic;
-using System.Text;
 
 namespace Squadron
 {
-    internal static class UniqueNameGenerator
+    public static class UniqueNameGenerator
     {
-        internal static string Create(string prefix)
+        public static string Create(string prefix)
```
Comment (lpzimmermann) [inline src/Core/UniqueNameGenerator.cs:7]: Adapt format from other methods

File: src/Mongo/MongoResource.cs [hunk 5]
```diff
@@ -57,22 +61,29 @@ private MongoClient GetClient()
...
         /// </summary>
         /// <value>The mongo database client.</value>
         public virtual IMongoClient Client => _client;
 
         /// <summary>
-        /// Gets the mongo database connection string.
+        /// Gets the external mongo database connection string that is exposed to the host
         /// </summary>
-        public string ConnectionString { get; private set; }
+        public string ExternalConnectionString { get; private set; }
```
Comment (RohrerF) [inline src/Mongo/MongoResource.cs:80]: @lpzimmermann Revert to make this not breaking

**Comment by glucaci** [comment]
Could you please revert all formatting changes, it will be much easier for review. We can do afterwards a PR only for formatting.

File: src/Mongo/MongoResource.cs [hunk 6]
```diff
@@ -58,22 +62,29 @@ private MongoClient GetClient()
...
-        /// Gets the mongo database connection string.
+        /// Gets the external mongo database connection string that is exposed to the host
         /// </summary>
         public string ConnectionString { get; private set; }
 
+        /// <summary>
+        /// Gets the internal mongo database connection string
+        /// that is exposed to the container network
+        /// </summary>
+        public string NetworkConnectionString { get; private set; }
```
Comment (philbir) [inline src/Mongo/MongoResource.cs:86]: Name is confusing. Should we rename to `InternalConnectionString`

**Pull request closed without merging**